### ISSUE TITLE
Report and evaluation

### ISSUE THREAD

**Issue by mkutu** [precision-sustainable-ag/Field-SegmentationTraining#44]
- [ ] Define and document a “golden” test set (diverse, high-quality masks)
- [x] Measure IoU, track train/val loss & IoU over epochs
- [x] Tabulate key hyperparameters and auto-generate summary report

**Comment by mkutu** [comment]
@mkutu

**Comment by sbaghba** [comment]
I will work on the third point (Tabulate key hyperparameters and auto-generate summary report) and show preliminary results and we can adjust it from there.
